nixos/modules/services/networking/firewalld/settings.nix

```diff
@@ -0,0 +1,202 @@
+          type =
+            let
+              policy = enum [
+                "DROP"
+                "REJECT"
+                "ACCEPT"
+              ];
+            in
+            either policy commas;
+          description = "The policy during reload.";
+          default = "INPUT:DROP,FORWARD:DROP,OUTPUT:DROP";
+        };
+        RFC3964_IPv4 = mkOption {
+          type = bool;
+          description = ''
+            Whether to filter IPv6 traffic with 6to4 destination addresses
+            that correspond to IPv4 addresses that should not be routed over
+            the public internet.
+          '';
+          default = true;
+        };
+        StrictForwardPorts = mkOption {
+          type = bool;
+          description = ''
+            If enabled, the generated destination NAT (DNAT) rules will NOT
+            accept traffic that was DNAT'd by other entities, e.g. docker.
+            Firewalld will be strict and not allow published container ports
+            until they're explicitly allowed via firewalld. If set to no, then
+            docker (and podman) integrates seamlessly with firewalld. Published
+            container ports are implicitly allowed.
+          '';
+          default = false;
+        };
+        NftablesFlowtable = mkOption {
+          type = separatedString " ";
+          description = ''
+            This may improve forwarded traffic throughput by enabling nftables
+            flowtable. It is a software fastpath and avoids calling nftables
+            rule evaluation for data packets. Its value is a space separate
+            list of interfaces.
+          '';
+          default = "off";
+        };
+        NftablesCounters = mkOption {
+          type = bool;
+          description = "Whether to add a counter to every nftables rule.";
+          default = false;
+        };
+        NftablesTableOwner = mkOption {
+          type = bool;
+          description = ''
+            If enabled, the generated nftables rule set will be owned
+            exclusively by firewalld. This prevents other entities from
+            mistakenly (or maliciously) modifying firewalld's rule set. If you
+            intend to modify firewalld's rules, set this to false.
+          '';
+          default = true;
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.settings.FirewallBackend == "nftables" -> config.networking.nftables.enable;
+        message = ''
+          FirewallD uses nftables as the firewall backend (by default), but
+          nftables support isn't enabled. Please read the description of
+          networking.nftables.enable for possible problems. If using nftables
+          is not desired, set services.firewalld.settings.FirewallBackend to
+          "iptables", but be aware that FirewallD has deprecated support for
+          it, and will override firewall rule set by other services, if any.
```

Comment:
Note, unlike option descriptions that are rendered by markdown, warnings and assertions get printed to stderr as-is.

Therefore we may not want such aggressive line wrapping, and may want to consider how we can structure sentences so that they don't need wrapping. Or can be wrapped semantically (i.e. at punctuation).

Not blocking, just a heads up.